components/brave_extension/extension/brave_extension/content_cosmetic.ts

```diff
@@ -249,41 +261,73 @@ const isSubTreeFirstParty = (elm: Element, possibleQueryResult?: IsFirstPartyQue
     return false
   }
   const htmlElement = asHTMLElement(elm)
-  if (!htmlElement || !htmlElement.innerText.trim().length) {
+  if (!htmlElement || isAdText(htmlElement.innerText) === false) {
     return false
   }
   return true
 }
 
-const hideSelectors = (selectors: string[]) => {
-  if (selectors.length === 0) {
+const unhideSelectors = (selectors: Set<string>) => {
+  if (selectors.size === 0) {
     return
   }
-
-  chrome.runtime.sendMessage({
-    type: 'hideThirdPartySelectors',
-    selectors
-  })
+  const origRuleCount = cosmeticStyleSheet.rules.length
+  // Find selectors we have a rule index for
+  const rulesToRemove = Array.from(selectors)
+    .map(selector =>
+      allSelectorsToRules.get(selector)
+    )
+    .filter(i => i !== undefined)
```

Comment:
what is this checking?  I dont see why this could happen

Reply:
if `allSelectorsToRules` is passed a selector it does not contain then it will give us an `undefined` entry, which is going to mess with knowing the count to shift other indexes, as well as we'll have to check for it anyway before attempting to remove the `undefined` index rule.
We may know this cannot happen, but this function does not, and if we pass `undefined` to `CSSStyleSheet.deleteRule()` then it's not going to delete the rule we intended and all our indexes will be off.